### PR TITLE
refactor: packager inspect command

### PR DESCRIPTION
**Pull request by schristoff**
relates to #2969

### CODE REVIEW

File: src/pkg/packager2/inspect.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+			imageList = append(imageList, component.Images...)
+		}
+		if len(imageList) > 0 {
+			imageList = helpers.Unique(imageList)
+			return imageList, nil
+		}
+		return nil, fmt.Errorf("failed listing images: list of images found in components: %d", len(imageList))
+	}
+
+	//todo: (schristoff) ?????
```
Comment (phillebaba) [inline src/pkg/packager2/inspect.go:61]: ??????

Comment (schristoff) [reply src/pkg/packager2/inspect.go]: Thank you, I resolved this immediately. :)

File: src/internal/packager2/inspect.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+	"github.com/defenseunicorns/pkg/helpers/v2"
+	"github.com/zarf-dev/zarf/src/api/v1alpha1"
+	"github.com/zarf-dev/zarf/src/internal/packager/sbom"
+	"github.com/zarf-dev/zarf/src/pkg/layout"
+)
+
+// ZarfInspectOptions tracks the user-defined preferences during a package inspection.
+type ZarfInspectOptions struct {
+	// PackagePaths
```
Comment (schristoff) [inline src/internal/packager2/inspect.go:19]: // PackagePaths

File: src/pkg/packager/common.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ type Packager struct {
 	variableConfig *variables.VariableConfig
 	state          *types.ZarfState
 	cluster        *cluster.Cluster
-	layout         *layout.PackagePaths
```
Comment (schristoff) [inline src/pkg/packager/common.go:37]: You can revert all this

File: src/internal/packager2/inspect.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			imageList = helpers.Unique(imageList)
+			return imageList, nil
+		}
+		return nil, fmt.Errorf("failed listing images: list of images found in components: %d", len(imageList))
+	}
+
+	return imageList, err
+}
+
+func getPackageMetadata(_ context.Context, layout *layout.PackagePaths) (v1alpha1.ZarfPackage, error) {
```
Comment (phillebaba) [inline src/internal/packager2/inspect.go:67]: NIT: do we need to pass context if it is not used in a private function?

File: src/test/e2e/11_oci_pull_inspect_test.go [hunk 5]
```diff
@@ -70,7 +70,7 @@ func (suite *PullInspectTestSuite) Test_1_Remote_Inspect() {
 	// Test inspect on a public package.
 	// NOTE: This also makes sure that Zarf does not attempt auth when inspecting a public package.
 	ref := fmt.Sprintf("oci://ghcr.io/zarf-dev/packages/dos-games:1.0.0-%s", e2e.Arch)
-	_, stdErr, err = e2e.Zarf(suite.T(), "package", "inspect", ref)
+	_, stdErr, err = e2e.Zarf(suite.T(), "package", "inspect", ref, "--insecure")
```
Comment (phillebaba) [inline src/test/e2e/11_oci_pull_inspect_test.go:73]: Which insecure feature are we using here? Insecure is now deprecated so we should aim to use the specific flag that we want to set to insecure.

File: src/test/e2e/31_checksum_and_signature_test.go [hunk 6]
```diff
@@ -23,15 +23,14 @@ func TestChecksumAndSignature(t *testing.T) {
...
-	require.NoError(t, err, stdOut, stdErr)
-
 	// Test that we don't get an error when we remember to provide the public key
 	stdOut, stdErr, err = e2e.Zarf(t, "package", "inspect", pkgName, publicKeyFlag)
 	require.NoError(t, err, stdOut, stdErr)
-	require.Contains(t, stdErr, "Verified OK")
+
+	/* Test operations during package inspect */
+	// Test that we can inspect the yaml of the package without the private key
+	stdOut, stdErr, err = e2e.Zarf(t, "package", "inspect", pkgName, "--insecure")
```
Comment (phillebaba) [inline src/test/e2e/31_checksum_and_signature_test.go:32]: Same with this and the other tests.

File: src/test/e2e/11_oci_pull_inspect_test.go (hunk 5 above)
Comment (schristoff) [reply src/test/e2e/11_oci_pull_inspect_test.go:73]: I will make it the longer one.

**Review by phillebaba — APPROVED** [review]
Good Job!

**Pull request merged**